core/pkg/filter/k8sobject/parser.go

```diff
@@ -0,0 +1,43 @@
+package k8sobject
+
+import (
+	"github.com/opencost/opencost/core/pkg/filter/ast"
+)
+
+// a slice of all the allocation field instances the lexer should recognize as
+// valid left-hand comparators
+var k8sObjectFilterFields []*ast.Field = []*ast.Field{
+	ast.NewField(FieldNamespace),
+	ast.NewField(FieldControllerName, ast.FieldAttributeNilable),
```

Comment:
Just wanted to point out (if you haven't noted field attributes yet) that `ast.FieldAttributeNilable` (or any attribute that might be needed) on a field is simply metadata that can be accessed in the compiler or code generator to help alleviate any special cases that may exist.

Reply:
Attributes are just bitflags (sorry, I think I'm showing my age), so they're not super expansive. I think Nilable starts at the 5th bit, and we're using an `int`, so it's expandable to another 27 custom attributes 🤷

Reply:
Oh very neat, I just copied whatever Allocation was using for consistency; I _think_ it also makes sense here.